docs/docs/apis/providers/supplier-interfaces-api.md

```diff
@@ -0,0 +1,138 @@
+
+## Key Features
+
+- **Extensible Architecture**: Define custom data provider implementations
+- **Interface Standardization**: Consistent contracts for different data sources
+- **Provider Flexibility**: Switch between implementations seamlessly
+- **Testing Support**: Mock implementations for unit and integration testing
+- **Service Composition**: Combine multiple suppliers for complex scenarios
+- **Custom Integration**: Integrate with proprietary or specialized data sources
+
+## Core Classes
+
+### Supplier Interfaces
+- `UtxoSupplier` - Provides UTXO data for addresses and transactions
+- `ProtocolParamsSupplier` - Supplies current protocol parameters
+- `TransactionProcessor` - Handles transaction submission and evaluation
+- `ScriptSupplier` - Provides Plutus scripts by hash
+- `TransactionEvaluator` - Evaluates transaction execution costs
+
+### Built-in Implementations
+- `OgmiosProtocolParamSupplier` - Ogmios-based protocol parameters
+- `OgmiosTransactionProcessor` - Ogmios transaction processing
+- `KupoUtxoSupplier` - Kupo-based UTXO provider
+- `OgmiosTransactionEvaluator` - Ogmios script evaluation
+
+## Usage Examples
+
+### Custom UtxoSupplier Implementation
+
+Implement a custom UTXO supplier:
+
+```java
+public class CustomUtxoSupplier implements UtxoSupplier {
+    
+    @Override
+    public List<Utxo> getPage(String address, Integer nrOfItems, Integer page, OrderEnum order) {
+        // Implement UTXO fetching logic
+        return fetchUtxosFromCustomSource(address, nrOfItems, page, order);
+    }
+    
+    @Override
+    public Optional<Utxo> getTxOutput(String txHash, int outputIndex) {
+        // Implement transaction output retrieval
+        return fetchTransactionOutput(txHash, outputIndex);
+    }
+    
+    @Override
+    public List<Utxo> getAll(String address) {
+        // Fetch all UTXOs for address
+        return fetchAllUtxosFromSource(address);
+    }
+}
+```
+
+### Custom ProtocolParamsSupplier
+
+Implement protocol parameters supplier:
+
+```java
+public class CustomProtocolParamsSupplier implements ProtocolParamsSupplier {
+    
+    @Override
+    public ProtocolParams getProtocolParams() {
+        // Fetch current protocol parameters from your source
+        return ProtocolParams.builder()
+            .minFeeA(44)
+            .minFeeB(155381)
+            .maxBlockSize(65536)
+            .maxTxSize(16384)
+            .keyDeposit(BigInteger.valueOf(2000000))
+            .poolDeposit(BigInteger.valueOf(500000000))
+            .build();
+    }
+}
+```
+
+### Using Custom Suppliers
+
+Integrate custom suppliers with QuickTx:
+
+```java
+// Create custom suppliers
+UtxoSupplier customUtxoSupplier = new CustomUtxoSupplier();
+ProtocolParamsSupplier protocolParamsSupplier = new CustomProtocolParamsSupplier();
+
+// Use with QuickTx
+QuickTxBuilder quickTxBuilder = new QuickTxBuilder(customUtxoSupplier)
```

Comment:
Can you please verify if fluent API support is available? I think you can only provide these suppliers through the QuickTxBuilder constructor.